### PR TITLE
Slightly different weekend treatment in TARGET calendar && removed 31.12. from TARGET calendar 

### PR DESCRIPTION
**Pull request by NiklasRodi**
## What is this PR for?
Slightly different weekend treatment in TARGET calendar:
- Assume you have a NEWYORK calendar that also adds WEEKDAY automatically as base calendar. Now you want to built union of TARGET and NEWYORK calendars by building a NEWYORK calendar with TARGET as base. Then you have NEWYORK with base WEEKDAY with base TARGET with base WEEKDAY. That is a bit ugly
- This is just a cosmetic change. Weekends will still be non-bussinesdays in the TARGET calendar

Removed 31.12. from TARGET calendar

### What type of PR is it?
Improvement/Feature

### CODE REVIEW

File: src/main/java6/net/finmath/time/businessdaycalendar/BusinessdayCalendarExcludingTARGETHolidays.java [hunk 1]
```diff
@@ -45,7 +44,9 @@ public boolean isBusinessday(LocalDate date) {
 		LocalDate datePlus2 = date.plusDays(2);
 		LocalDate dateBefore = date.minusDays(1);
 
-		return	(baseCalendar == null || baseCalendar.isBusinessday(date))
+		BusinessdayCalendarInterface weekdayCalendar = new BusinessdayCalendarExcludingWeekends();
```
Comment (cfries) [inline src/main/java6/net/finmath/time/businessdaycalendar/BusinessdayCalendarExcludingTARGETHolidays.java]: Construction of a new object in every call of isBusinessday may be a performance issue.

**Comment by cfries** [comment]
However: I assume that the && operator ensures that the test for weekend is performed only once, regardless of how many base calendars are excluding the weekends.

**Comment by NiklasRodi** [comment]
That is then also true for my approach. I'm e.g. thinking about the debugging case. If you have a {EUR,USD,GBP} calendar I would like to see something like GBP with base USD with base EUR and not GBP with base WEEKDAY with base USD with base WEEKDAY with base EUR with base WEEKDAY. The latter makes it very hard to understand the situation, doesn't it?

File: src/main/java6/net/finmath/time/businessdaycalendar/BusinessdayCalendarExcludingTARGETHolidays.java (hunk 1 above)
Comment (NiklasRodi) [inline src/main/java6/net/finmath/time/businessdaycalendar/BusinessdayCalendarExcludingTARGETHolidays.java]: How about I add the member "private BusinessdayCalendarInterface weekdayCalendar = new BusinessdayCalendarExcludingWeekends()" to the TARGET calendar?

Comment (NiklasRodi) [inline src/main/java6/net/finmath/time/businessdaycalendar/BusinessdayCalendarExcludingTARGETHolidays.java]: even better "private final ..." :-)

**Comment by cfries** [comment]
Yes. And make it final.

**Comment by cfries** [comment]
The point with the debugging (that is the impact on toString()) is a good one!

LGTM.

**Pull request merged**